### ISSUE TITLE
Prediction failed predicting anything after training

### ISSUE THREAD

**Issue by yby026** [PatBall1/detectree2#225]
Hi developers:

We are trying to use the predict function to visualise the trained model performance, and when we were loading the trained model into the prediction function on a new imagery, it failed predicting anything. We initially trained the model several times based on our old data to adapt to our data, and then we used this trained model to train on our more recent data. With our previously trained model, we can produce outcome on the new data. But when we were testing the newly trained model, the prediction failed. Not sure what is triggering this, but it seems not because of the workflow, and seems the model inference started normally. 

The code we used for prediction follows the tutorial, with changing `crowns = stitch_crowns(tiles_path + "/predictions_geo/", 1) ` to `crowns = stitch_crowns(tiles_path + "/predictions_geo/", 0) `, as when the shift is 1, the code failed to during `stitch_crown` function.
```
from detectree2.preprocessing.tiling import tile_data 
from detectree2.models.outputs import project_to_geojson, stitch_crowns, clean_crowns 
from detectree2.models.predict import predict_on_data 
from detectree2.models.train import setup_cfg 
from detectron2.engine import DefaultPredictor 
import rasterio 
import os 
site_path = "/nesi/nobackup/aut04447/data/prediction/nina/rgb_2025" 
img_path = site_path + "/nina_2025_2pt3_clip.tif" 
tiles_path = site_path + "/_tile2" 
model_path = "/nesi/nobackup/aut04447/models/260216_base_delineation_b_2/model_31.pth" # this model is trained on Boyle updated species annotation file, the pretrained model is trained on several old AUT UAV images and a aerial image clip 
print(f"Using model: {model_path}") 
cfg = setup_cfg(update_model=model_path) 
print(f"Number of classes: {cfg.MODEL.ROI_HEADS.NUM_CLASSES}") 
print(f"Model weights: {cfg.MODEL.WEIGHTS}") 
predict_on_data(tiles_path, predictor=DefaultPredictor(cfg), multiclass=False) 
project_to_geojson(tiles_path, tiles_path + "/predictions/", tiles_path + "/predictions_geo/") 
crowns = stitch_crowns(tiles_path + "/predictions_geo/", 0) 
if len(crowns) == 0: 
  print("\n!!! WARNING: No crowns detected !!!") 
  print("Checking tile input...") 
  tile_files = [f for f in os.listdir(tiles_path) if f.endswith(('.png', '.jpg', '.tif'))] 
  print(f"Number of tile files: {len(tile_files)}") 
  if tile_files: 
     print(f"Sample tile: {tile_files[0]}") 
  import sys 
  sys.exit(1) 
clean = clean_crowns(crowns, 0.6, confidence = 0.3 ) # set a confidence>0 to filter out less confident crowns 
clean = clean.set_geometry(clean.simplify(0.1)) 
clean.to_file(site_path + "/baseline_BoylePretrain_NinaCrowns_out.gpkg")
```

The log is attached here:
[baseline_boyleModel_nina2025_detectree_train_4902037.txt](https://github.com/user-attachments/files/25812746/baseline_boyleModel_nina2025_detectree_train_4902037.txt)

And the outcome file contains 0 feature
<img width="602" height="31" alt="Image" src="https://github.com/user-attachments/assets/b30e00ee-d1ef-4160-9700-a640c345216d" />

The training curve of this newly trained model is shown here:
<img width="579" height="453" alt="Image" src="https://github.com/user-attachments/assets/c256eb19-c9ef-4c84-849f-44a61cb4846f" />

Platform: NeSI HPCs

**Comment by PatBall1** [comment]
Thanks for the detailed report. A few things to investigate:

1. **Check training loss convergence**: Your training curve shows the loss is still decreasing at iteration 31 -- the model may not have converged sufficiently, or it may have overfit to the training data and lost generalization. Try training for more iterations or using early stopping (`MyTrainer` with `patience` parameter).

2. **Lower the confidence threshold**: The `clean_crowns(crowns, 0.6, confidence=0.3)` call filters out crowns below 0.3 confidence. Try `confidence=0.0` first to see if the model is producing any predictions at all, just at low confidence.

3. **Check predictions folder**: Before `stitch_crowns`, check if `predictions_geo/` contains any non-empty geojson files. If they're all empty, the issue is at the model inference stage.

4. **`stitch_crowns` shift parameter**: Using `shift=0` instead of `shift=1` works around a tile naming convention mismatch. This suggests your tiles may have been generated with different settings than expected. The shift parameter controls how many characters to skip when parsing coordinates from tile filenames.

5. **Model compatibility**: If you retrained on new data starting from a previously fine-tuned model, ensure the number of classes matches between the base model and the new training run.

**Comment by yby026** [comment]
Hi James, I checked all the logs from previous trainings and found the issue was not with the training pipeline, but because I used group normalisation but in prediction it didn't extract the normalisation method from cfg. Thank you helping!
